examples/text/plot_document_classification_20newsgroups.py

```diff
@@ -50,248 +42,388 @@
+        )
+        print(f"{len(data_test.data)} documents - {data_test_size_mb:.2f}MB (test set)")
+        print("%d categories" % len(target_names))
+        print(
+            f"vectorize training done in {duration_train:.3f}s "
+            f"at {data_train_size_mb / duration_train:.3f}MB/s"
+        )
+        print("n_samples: %d, n_features: %d" % X_train.shape)
+        print(
+            f"vectorize testing done in {duration_test:.3f}s "
+            f"at {data_test_size_mb / duration_test:.3f}MB/s"
+        )
+        print("n_samples: %d, n_features: %d" % X_test.shape)
+
+    return X_train, X_test, y_train, y_test, feature_names, target_names
 
 
-data_train_size_mb = size_mb(data_train.data)
-data_test_size_mb = size_mb(data_test.data)
+# %%
+# Compare feature effects
+# -----------------------
+# We train a first classification model without attempting to strip the metadata
+# of the dataset.
 
-print(
-    "%d documents - %0.3fMB (training set)" % (len(data_train.data), data_train_size_mb)
+X_train, X_test, y_train, y_test, feature_names, target_names = load_dataset(
+    verbose=True
 )
-print("%d documents - %0.3fMB (test set)" % (len(data_test.data), data_test_size_mb))
-print("%d categories" % len(target_names))
 
 # %%
-# Vectorize the training and test data
-# -------------------------------------
-#
-# split a training set and a test set
-y_train, y_test = data_train.target, data_test.target
+# Our first model is an instance of the
+# :class:`~sklearn.linear_model.RidgeClassifier` class. This is a linear
+# classification model that uses the mean squared error on {-1, 1} encoded
+# targets, one for each possible class. Contrary to
+# :class:`~sklearn.linear_model.LogisticRegression`, `RidgeClassifier` does not
+# provide probabilistic predictions (no `predict_proba` method), but it is often
+# faster to train.
+
+from sklearn.linear_model import RidgeClassifier
+
+clf = RidgeClassifier(tol=1e-2, solver="sparse_cg")
+clf.fit(X_train, y_train)
+pred = clf.predict(X_test)
 
 # %%
-# Extracting features from the training data using a sparse vectorizer
-from time import time
+# We plot the confusion matrix of this classifier to find if there is a pattern
+# in the classification errors.
 
-from sklearn.feature_extraction.text import TfidfVectorizer
-from sklearn.feature_extraction.text import HashingVectorizer
+import matplotlib.pyplot as plt
+from sklearn.metrics import ConfusionMatrixDisplay
+
+fig, ax = plt.subplots(figsize=(10, 5))
+ConfusionMatrixDisplay.from_predictions(y_test, pred, ax=ax)
+ax.xaxis.set_ticklabels(target_names)
+ax.yaxis.set_ticklabels(target_names)
+_ = ax.set_title(
+    f"Confusion Matrix for {clf.__class__.__name__}\non the original documents"
+)
+
+# %%
+# The confusion matrix highlights that documents of the `alt.atheism` class are
+# often confused with documents with the class `talk.religion.misc` class while
+# and vice-versa which is expected since the topics are semantically related.
```

Comment:
```suggestion
# The confusion matrix highlights that documents of the `alt.atheism` class are
# often confused with documents with the class `talk.religion.misc` class and
# vice-versa which is expected since the topics are semantically related.
```